core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -170,6 +170,15 @@
         return new ConcatArrayStreamMessage<>(streamMessages);
     }
 
+    /**
+     * Returns a concatenated {@link StreamMessage} which relays items of the specified {@link Publisher} of
+     * {@link Publisher}s in order, non-overlappingly, one after the other finishes.
+     */
+    static <T> StreamMessage<T> concat(Publisher<? extends Publisher<? extends T>> publishers) {
```

Comment:
`concat` is an official operator name that emits items one after the other without interleaving them.
http://reactivex.io/documentation/operators/concat.html
However flat(flatten/merge) will be used to merge all items concurrently.
http://reactivex.io/documentation/operators/merge.html

Reply:
JFYI, `flatten` is linked to merge in this page.
http://reactivex.io/documentation/operators.html

Reply:
> concat is an official operator name that emits items one after the other without interleaving them.

Yeah, we use `concat` when the parameters are `Publisher<? extends T>...` and `Iterable<? extends Publisher<? extends T>>` which means that all the publishers are at the same level so they are concatenated.
I thought this case is a bit different because it seems like `Publihser`s are the parent and child relationship. So I thought we need a different name. `flat` was not a good name though. 😅

Reply:
I think it is not delimited to an array of `Publisher`s. Other Reactive Streams implementations already choose `concat`. 

https://github.com/reactor/reactor-core/blob/8e8262f273e8b82856c5a4bf9e63f8ab36d78876/reactor-core/src/main/java/reactor/core/publisher/Flux.java#L452
https://github.com/ReactiveX/RxJava/blob/e46ce76b95653006a709f230a71768a731c7d830/src/main/java/io/reactivex/rxjava3/core/Observable.java#L1094

Reply:
Thanks for the link. Yeah I guess so. 😄